### PR TITLE
Replace Travis CI with GitHub Actions

### PR DESCRIPTION
**Pull request by lzaoral**
Replace Travis CI with GitHub Actions and fix all issues that were uncovered during this process.

### CODE REVIEW

**Comment by lzaoral** [comment]
Some jobs depend on https://github.com/VeriFIT/ProStatA/pull/13.

**Comment by lzaoral** [comment]
@kdudka, could you please take a look at the `Ubuntu 18.04 (compiler: gcc, host: gcc)` test failures? Both compiler and host are GCC 7 which should work since at was a tested configuration in Travis: 
https://github.com/kdudka/predator/blob/8b5e8b31250b3d0ecd3538a6de2d06b507de5f9d/.travis.yml#L106-L117

**Comment by kdudka** [comment]
`build-aux/gcc-8.3.0.patch ` is not applied because the `matrix.host == 'gcc-7'` is wrong.

Should the condition rather be spelled as `matrix.version == '18.04'`?

File: sl/symheap.cc [hunk 1]
```diff
@@ -1928,8 +1928,6 @@ SymHeapCore::SymHeapCore(TStorRef stor, Trace::Node *trace):
     stor_(stor),
     d(new Private(trace))
 {
-    CL_BREAK_IF(!&stor_);
```
[the remark this answers is not in the document]
Comment (lzaoral) [reply sl/symheap.cc]: Addressed in 8387e04f3fdb020d79d95e9039dc9da601352262.

[the remark this answers is not in the document]
Comment (kdudka) [reply sl/symheap.cc]: Looks good.  Thanks!

**Review by kdudka — APPROVED** [review]
Looks perfect.  Thank you for making the CI work again and extending it!  We should merge #62 first and rebase this pull request on top of it.

**Comment by lzaoral** [comment]
I've just rebased this PR with #62 so if CI remains green, these PRs can be merged together.

**Pull request merged**